index.js

```diff
@@ -2984,37 +2984,55 @@
 
   //. ### Array
 
-  //# append :: a -> Array a -> Array a
+  //# append :: (Applicative f, Semigroup (f a)) => a -> f a -> f a
```

Comment:
> If `f` is not a Semigroup, wouldn't that mean by extension that `f a` isn't a Semigroup either?

Concrete example might help as well:

```haskell
Integer -- is not a Semigroup (no clear way of "combining" Integers)

Just Integer -- is not a Semigroup (we can "combine" Maybe's, but not Integers)

Just List Integer -- is a Semigroup because List of anything is a Semigroup
```

Just `Semigroup f` isn't enough because we can't know if we can "combine" specific things "wrapped" in `f` (i.e. @safareli's comment above), and `Semigroup a` is not enough because `f` might not define how `f`s should be "combined".